### ISSUE TITLE
Adding a channel fails if a package is not found

### ISSUE THREAD

**Issue by AlexisTM** [prefix-dev/pixi#2517]
### Checks

- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the [latest version](https://github.com/prefix-dev/pixi/releases) of pixi, using `pixi --version`.

### Reproducible example

```shell
pixi add non-existing-package
pixi project channel add robostack
```

### Issue description

The robotstack channel is not added while the error is related to the non-existing-package

### Expected behavior

The robotstack channel is added, as it may be missing for that package to be found

**Comment by ruben-arts** [comment]
Thanks for the report, this was by design but I think we can improve the UX. 

It's currently designed to:
- Validate the channel
- Try to solve the environment with this channel included (as it might fail because of channel priority rules)
- Install the environment to get an up to date setup
- If all succeeds add the channel to the manifest and save.

This all assumes a working manifest as input, and then not creating a broken result. 

A few ideas:
- Break design and just add the channel to the file without checking
- Test a before and after adding and based on that add it or not. 
- Give a dialog that asks the user if they would still like to add it while it does result in a broken state.

**Comment by AlexisTM** [comment]
Indeed, I didn't think about adding a channel may break the build.
In my case it came from #2516 which does add packages regardless if they exists.

**Comment by AlexisTM** [comment]
Closed as it is a feature and #2516 is the bug.
